src/base/bittorrent/torrenthandle.h

```diff
@@ -465,6 +465,7 @@ namespace BitTorrent
         QHash<QString, TrackerInfo> m_trackerInfos;
 
         bool m_started = false;
+        bool m_didCheck = false;
```

Comment:
It seems like you don't really need this global flag. You need to set some local flag in "handleTorrentFinishedAlert()" before "updateStatus()" call.

Reply:
Nice work! Will you provide a fix then?

Reply:
And fix your commit message according to Coding Guidelines, please.

Reply:
>Nice work! Will you provide a fix then?

Well, I've done it myself.